homeassistant/components/egauge/sensor.py

```diff
@@ -0,0 +1,103 @@
+"""Sensor platform for eGauge energy monitors."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from egauge_async.json.models import RegisterType
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import UnitOfEnergy, UnitOfPower
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .coordinator import EgaugeConfigEntry, EgaugeData, EgaugeDataCoordinator
+from .entity import EgaugeEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class EgaugeSensorEntityDescription(SensorEntityDescription):
+    """Extended sensor description for eGauge sensors."""
+
+    native_value_fn: Callable[[EgaugeData, str], float]
+    available_fn: Callable[[EgaugeData, str], bool]
+
+
+SENSORS: tuple[EgaugeSensorEntityDescription, ...] = (
+    EgaugeSensorEntityDescription(
+        key="power",
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfPower.WATT,
+        native_value_fn=lambda data, register: data.measurements[register],
+        available_fn=lambda data, register: register in data.measurements,
+    ),
+    EgaugeSensorEntityDescription(
+        key="energy",
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        native_unit_of_measurement=UnitOfEnergy.JOULE,
+        suggested_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        native_value_fn=lambda data, register: data.counters[register],
+        available_fn=lambda data, register: register in data.counters,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: EgaugeConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up eGauge sensor platform."""
+    coordinator = entry.runtime_data
+    sensors: list[SensorEntity] = []
+
+    async_add_entities(
+        EgaugeSensor(coordinator, register_name, sensor)
+        for sensor in SENSORS
+        for register_name, register_info in coordinator.data.register_info.items()
+        if register_info.type == RegisterType.POWER
+    )
+
+    async_add_entities(sensors)
```

Comment:
```suggestion

    async_add_entities(
        EgaugeSensor(coordinator, register_name, sensor)
        for sensor in SENSORS
        for register_name, register_info in coordinator.data.register_info.items()
        if register_info.type == RegisterType.POWER
    )

```
looks like there's a leftover